x-pack/plugins/enterprise_search/server/__mocks__/router.mock.ts

```diff
@@ -68,7 +68,14 @@ export class MockRouter {
   public validateRoute = (request: MockRouterRequest) => {
     if (!this.payload) throw new Error('Cannot validate wihout a payload type specified.');
 
-    const [config] = this.router[this.method].mock.calls[0];
+    // @ts-ignore
```

Comment:
This was a fix for an issue I mentioned on Slack. Sorry, I should have put this in a separate commit.

Validation checks were failing if you have validation on multiple endpoints that used the same method.

i.e.,

PUT /api/app_search/engines/{engineName}/search_settings
PUT /api/app_search/engines/{engineName}/search_settings_search

This is because the prior code was always using the validation from the first registered put route: `this.router[this.method].mock.calls[0];`. The code now checks the method and path.

Reply:
I'm ignoring TS because it was complaining and I didn't think it was worth attempting to fix in a mock like this.

Reply:
Can you add this explanation as a comment? I think there's also a syntax to start and end a ts ignore block, which might be better to wrap lines 72-74 in

Reply:
1f9879d08d78793d273437013ba38ca382f56ea0